io/eolearn/io/processing_api.py

```diff
@@ -0,0 +1,205 @@
+        self.additional_data = additional_data or []
+
+        self.all_bands = self.bands + [f_name for _, f_name in self.additional_data]
+
+    @staticmethod
+    def request_from_date(request, date, maxcc, time_difference):
+        ''' Make a deep copy of a request and sets it's (from, to) range according to the provided 'date' argument
+
+        :param request: Path to cache_folder. If set to None (default) requests will not be cached.
+        :type request: str
+        '''
+
+        date_from, date_to = date - time_difference, date + time_difference
+        time_from, time_to = date_from.isoformat() + 'Z', date_to.isoformat() + 'Z'
+
+        request = deepcopy(request)
+        for data in request['input']['data']:
+            time_range = data['dataFilter']['timeRange']
+            time_range['from'] = time_from
+            time_range['to'] = time_to
+
+            # this should be moved to sentinelhub-py package, it was done here to avoid doing another release of sh-py
+            data['dataFilter']['maxCloudCoverage'] = int(maxcc * 100)
+
+        return request
+
+    def generate_evalscript(self):
+        ''' Generate the evalscript to be passed with the request, based on chosen bands
+        '''
+        evalscript = """
+            function setup() {{
+                return {{
+                    input: [{{
+                        bands: {bands},
+                        units: "DN"
+                    }}],
+                    output: {{
+                        id:"default",
+                        bands: {num_bands},
+                        sampleType: SampleType.UINT16
+                    }}
+                }}
+            }}
+
+            function updateOutputMetadata(scenes, inputMetadata, outputMetadata) {{
+                outputMetadata.userData = {{ "norm_factor":  inputMetadata.normalizationFactor }}
+            }}
+
+            function evaluatePixel(sample) {{
+                return {samples}
+            }}
+        """
+
+        samples = ', '.join(['sample.{}'.format(band) for band in self.all_bands])
+        samples = '[{}]'.format(samples)
+
+        return evalscript.format(bands=json.dumps(self.all_bands), num_bands=len(self.all_bands), samples=samples)
+
+    def get_dates(self, bbox, time_interval):
+        ''' Make a WebFeatureService request to get dates and clean them according to self.time_difference
+        '''
+        wfs = WebFeatureService(
+            bbox=bbox, time_interval=time_interval, data_source=self.data_source, maxcc=self.maxcc
+        )
+
+        dates = wfs.get_dates()
+
+        if len(dates) == 0:
+            raise ValueError("No available images for requested time range: {}".format(time_interval))
+
+        dates = sorted(dates)
+        dates = [dates[0]] + [d2 for d1, d2 in zip(dates[:-1], dates[1:]) if d2 - d1 > self.time_difference]
+        return dates
+
+    @staticmethod
+    def size_from_resolution(bbox, resolution):
```

Comment:
we are sure the bbox is in UTM, or do we need to add a check? This transformation won't hold for other CRSs

Reply:
Yes, I agree there should at least be a check, because this is an over-simplification that would only work in specific cases.

Reply:
Changed this as @AleksMat suggested.